instances.js

```diff
@@ -25,21 +25,14 @@ export async function get({id}) {
   return response.data;
 }
 
-export async function getAll({controller} = {}) {
-  const instanceIds = await _getInstanceIds({id: controller});
-  const promises = instanceIds.map(async id => {
-    try {
-      const instance = await get({id});
-      return instance;
-    } catch(e) {
-      console.log(`Unable to fetch issuer: "${id}"`);
-      console.error(e);
-    }
-  });
-  // Resolve promises and filter out undefined
-  // FIXME: Consider pulling in promises lib to limit concurrency
-  const instances = await Promise.all(promises);
-  return instances.filter(promise => promise);
+export async function getAll({controller}) {
```

Comment:
This code only works when the controller is the `controller` of the issuer instance. When we delegate ZCAPs to a User and that User uses this API call the code fails to retrieve all issuer instances. This is because that User is not the controller on the instance. Instead, we need to retrieve all the capabilities for the User that encodes data about the controller of the instance and then use that to retrieve the instance.

In fact, it seems that we are not always passing the `controller` of the instance to this API. This is why the behavior is different. One time we pass the `controller` of the instance and another time we pass the `controller` of the delegated capability to access the instance. Subtle differences but it is why the API was written the way it was.

Reply:
Sounds terrible.  Seems to me that there need to be intermediate APIs or something to disambiguate `controller`.  We're dealing with instances here, so I don't think we should be concerning ourselves with unpacking zcaps.

Reply:
@dlongley Thoughts?

Reply:
Is this PR still relevant? I'd like to close or merge it and then deal with it along with the other refactoring.

Reply:
I don't know, is getting a list of instances by `controller` still/going to be a thing?